app/models/users.rb

```diff
@@ -1,9 +1,29 @@
 class Users < ApplicationRecord
   has_many :comments, class_name: 'Comments', foreign_key: 'author_id'
-  has_many :likes, class_name: 'Likes', foreign_key: 'author_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'author_id'
   has_many :posts
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true }
+  validates :posts_counter, numericality: { minimum: 0 }
   # Ex:- scope :active, -> {where(:active => true)}
-  def most_recent_n(name = self, lim = 3)
-    Posts.where(author_id: name).order(created_at: :desc).limit(lim)
+  def most_recent_n(id = self.id, lim = 3)
+    Posts.where(author_id: id).order(created_at: :desc).limit(lim)
+  end
+
+  def count_posts(name = id)
+    posts = Posts.where(author_id: name)
+    posts.length
+  end
+
+  def get_all_posts(id = self.id)
+    posts = Posts.where(author_id: id)
+    dic = {}
+    count = 1
+    posts.each do |single|
+      dic[count] = single
+      count += 1
+    end
+
+    dic
```

Comment:
- [ ] Kindly note that you are having many errors in your modals.
- Names in convention Rails is really sensitive regarding names in convention 
. The Model should always be in the singular. The example from your (Like and comment) models.

That's why some of your tests are not passing kindly fixe them as suggested by the previous reviewer.

![image](https://user-images.githubusercontent.com/65068771/163693832-3dd41035-7e9a-4fd6-9e0b-bc734c24b5ed.png)